sys/net/include/fib/fib.h

```diff
@@ -0,0 +1,141 @@
+ *
+ * @{
+ *
+ * @file        fib.h
+ * @brief       Types and functions for FIB
+ * @author      Martin Landsmann
+ */
+
+#ifndef _FIB_H
+#define _FIB_H
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief Reactive Routing Protocol (RRP) message content to request/reply route discovery
+ */
+typedef struct rrp_address_msg_t {
+    uint8_t *address;      /**< The pointer to the address */
+    uint8_t address_size;  /**< The address size */
+    kernel_pid_t iface_id; /**< The interface used for the next-hop reply.
+                                If it is KERNEL_PID_UNDEF, no next-hop has been found. */
+    uint32_t lifetime;     /**< The lifetime of this next-hop */
+} rrp_address_msg_t;
+
+#define FIB_MSG_RRP_SIGNAL (0x99)              /**< message type for RRP notifications         */
+
+#define FIB_SUCCESS_EXACT_MATCH (1)            /**< return value for successfull exact match   */
+#define FIB_SUCCESS (0)                        /**< return value for success                   */
+#define FIB_ERROR (-1)                         /**< error value for generic failure            */
+#define FIB_ERROR_DESTINATION_UNREACHABLE (-2) /**< error value indicating unreachable dest.   */
+#define FIB_ERROR_INSUFICCIENT_OUT_SIZE (-3)   /**< error value for invalid output buffer size */
+
+/**
+ * @brief initializes all FIB entries with 0
+ */
+void fib_init(void);
+
+/**
+ * @brief de-initializes the FIB entries
+ */
+void fib_deinit(void);
+
+/**
+ * @brief Registration of reactive routing protocol handler function
+ */
+void fib_register_rrp(void);
+
+/**
+ * @brief Adds a new entry in the corresponding FIB table for global destination and next hop
+ *
+ * @param[in] iface_id       the interface ID
+ * @param[in] glob_dst       the destination address
+ * @param[in] glob_dst_size  the destination address size
+ * @param[in] next_hop       the next hop address to be updated
+ * @param[in] next_hop_size  the next hop address size
+ * @param[in] lifetime       the lifetime in ms to be updates
+ *
+ * @return FIB_SUCCESS on success
+ *         FIB_ERROR if the entry cannot be created due to unsufficient RAM
+ */
+int fib_table_add_entry(kernel_pid_t iface_id, uint8_t *glob_dst, size_t glob_dst_size,
+                        uint8_t *next_hop, size_t next_hop_size, uint32_t lifetime);
+
+/**
+ * @brief Updates an entry in the corresponding FIB table for global destination and next hop
+ *
+ * @param[in] iface_id       the interface ID
+ * @param[in] glob_dst       the destination address
+ * @param[in] glob_dst_size  the destination address size
+ * @param[in] next_hop       the next hop address to be updated
+ * @param[in] next_hop_size  the next hop address size
+ * @param[in] lifetime       the lifetime in ms to be updates
+ *
+ * @return FIB_SUCCESS on success
+ *         FIB_ERROR if the entry cannot be updated due to unsufficient RAM
+ */
+int fib_table_upd_entry(kernel_pid_t iface_id, uint8_t *glob_dst, size_t glob_dst_size,
```

Comment:
Was briefly confused, because I read `udp`. Can you name it `fib_table_update_entry`.